src/main/mondrian/rolap/aggmatcher/AggStar.java

```diff
@@ -1121,6 +1128,19 @@ public void print(final PrintWriter pw, final String prefix) {
                 pw.print("): ");
                 pw.print(generateRollupString(sqlQuery));
             }
+
+            @Override
+            public String generateExprString(SqlQuery query) {
+                String exprString = super.generateExprString(query);
+                RolapAggregator rollupAggregator = getRollupAggregator();
+                if (rollupAggregator instanceof BaseAggor) {
+                    BaseAggor agg = (BaseAggor) rollupAggregator;
```

Comment:
This is not a good pattern. Please see if you can make this better. Do we really need the downcast? Can it ever be a different class? Shouldn't the method be part of the RolapAggregator?